cli/tsc/99_main_compiler.js

```diff
@@ -752,11 +752,21 @@ delete Object.prototype.__proto__;
           /** @type {Record<string, any[]>} */
           const diagnosticMap = {};
           for (const specifier of request.specifiers) {
-            diagnosticMap[specifier] = fromTypeScriptDiagnostic([
-              ...languageService.getSemanticDiagnostics(specifier),
-              ...languageService.getSuggestionDiagnostics(specifier),
-              ...languageService.getSyntacticDiagnostics(specifier),
-            ].filter(({ code }) => !IGNORED_DIAGNOSTICS.includes(code)));
+            let diagnostics = [];
+            try {
+              diagnostics = [
+                ...languageService.getSemanticDiagnostics(specifier),
+                ...languageService.getSuggestionDiagnostics(specifier),
+                ...languageService.getSyntacticDiagnostics(specifier),
+              ];
+            } catch {
```

Comment:
Huh, interesting. Would it make sense to do this per method call? I guess not since these are transient errors? What kind of errors get thrown?

Reply:
It would be nice to know exactly what’s happening, but this seems fine to do.

Reply:
Looking at all the code, I suspect it is some sort of debug logging/failure that is occurring, related to an attempt to find a source file, but it being in a bad state it non-critically debug errors, but because we don't fully implement the debug host, instead of it being something meaningful handled by the TypeScript internal debugger, it bubbles up as a `{}`.

One thing we don't do is validate on the `getDiagnostics` that all the specifiers have source files... actually that is an interesting point, I will look into that.  It could be we just need to go grab any missing source files, where we are asking for diagnostics for source files that aren't in tsc yet.

Reply:
ok, found it, it was `ts.OperationCanceledException` being thrown, which we should just swallow like we do with the `OperationCanceledError`, which explains the transient nature of the error.  I have refactored it again to better handle that error.

Reply:
Perfect! Thanks for looking into it.